valhalla/sif/dynamiccost.h

```diff
@@ -359,10 +361,11 @@ class DynamicCost {
     // you have forward access for the mode you care about
     // you dont care about what mode has access so long as its forward
     // you dont care about the direction the mode has access to
-    return ((edge->forwardaccess() & access_mask_) ||
-            (ignore_access_ && (edge->forwardaccess() & baldr::kAllAccess)) ||
-            (ignore_oneways_ && (edge->reverseaccess() & access_mask_))) &&
-           (edge->use() != baldr::Use::kConstruction);
+    return (((edge->forwardaccess() & access_mask_) ||
+             (ignore_access_ && (edge->forwardaccess() & baldr::kAllAccess)) ||
+             (ignore_oneways_ && (edge->reverseaccess() & access_mask_))) &&
+            (edge->use() != baldr::Use::kConstruction)) ||
+           (ignore_constructions_ && edge->use() == baldr::Use::kConstruction);
```

Comment:
this one is confusing we should update the comment :smile: could it be a bit simpler, something like:
```suggestion
    return (((edge->forwardaccess() & access_mask_) ||
             (ignore_access_ && (edge->forwardaccess() & baldr::kAllAccess)) ||
             (ignore_oneways_ && (edge->reverseaccess() & access_mask_))) &&
           (ignore_constructions_ || edge->use() != baldr::Use::kConstruction);
```

Reply:
That is not correct. I did try that and noticed that it didn't work anymore. The reason being, that the parentheses are not on the same level as in the IsAccessible function. If a way has the construction tag, then both forward and reverse access is set to false in when building the graph in graph.lua: https://github.com/valhalla/valhalla/blob/master/lua/graph.lua#L1873-L1908

So what I do here is allow access regardless of the access, but only if it is a construction way and we are ignoring construction.